unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/BuilderEntity/DecentralandEntityToEdit.cs

```diff
@@ -0,0 +1,295 @@
+    {
+        DecentralandEntity.MeshesInfo meshInfo = entity.meshesInfo;
+        if (meshInfo == null || meshInfo.currentShape == null)
+            return;
+        if (!meshInfo.currentShape.IsVisible())
+            return;
+        if (!meshInfo.currentShape.IsVisible() && meshInfo.currentShape.HasCollisions())
+            return;
+        if (!meshInfo.currentShape.IsVisible() && !meshInfo.currentShape.HasCollisions())
+            return;
+
+        if (!collidersDictionary.ContainsKey(entity.scene.sceneData.id + entity.entityId))
+        {
+            if (entity.children.Count > 0)
+            {
+                using (var iterator = entity.children.GetEnumerator())
+                {
+                    while (iterator.MoveNext())
+                    {
+                        CreateCollidersForEntity(iterator.Current.Value);
+                    }
+                }
+            }
+
+
+            GameObject entityCollider = new GameObject(entity.entityId);
+            entityCollider.layer = LayerMask.NameToLayer("OnBuilderPointerClick");
+
+
+            for (int i = 0; i < meshInfo.renderers.Length; i++)
+            {
+                Transform t = entityCollider.transform;
+                t.SetParent(meshInfo.renderers[i].transform);
+                t.ResetLocalTRS();
+
+                var meshCollider = entityCollider.AddComponent<MeshCollider>();
+
+                if (meshInfo.renderers[i] is SkinnedMeshRenderer)
+                {
+                    Mesh meshColliderForSkinnedMesh = new Mesh();
+                    (meshInfo.renderers[i] as SkinnedMeshRenderer).BakeMesh(meshColliderForSkinnedMesh);
+                    meshCollider.sharedMesh = meshColliderForSkinnedMesh;
+                    t.localScale = new Vector3(1 / entity.gameObject.transform.lossyScale.x, 1 / entity.gameObject.transform.lossyScale.y, 1 / entity.gameObject.transform.lossyScale.z);
+                }
+                else
+                {
+                    meshCollider.sharedMesh = meshInfo.renderers[i].GetComponent<MeshFilter>().sharedMesh;
+                }
+                meshCollider.enabled = meshInfo.renderers[i].enabled;
+
+            }
+
+            collidersDictionary.Add(entity.scene.sceneData.id + entity.entityId, entityCollider);
+        }
+    }
+
+    void SetLockStatus(bool isLocked)
+    {
+        this.isLocked = isLocked;
+        onStatusUpdate?.Invoke(this);
+    }
+
+    void CheckIfEntityIsFloor()
+    {
+        if (rootEntity.meshesInfo == null || rootEntity.meshesInfo.currentShape == null)
+            return;
+        if (rootEntity.meshesInfo.renderers == null && rootEntity.meshesInfo.renderers.Length <= 0)
+            return;
+        if (rootEntity.meshesInfo.mergedBounds.size.y >= 0.05f)
+            return;
+        if (rootEntity.gameObject.transform.position.y >= 0.05f)
+            return;
+
+        SetLockStatus(true);
+    }
+
+    void CheckIfEntityIsVoxel()
```

Comment:
same above, `TestIfEntityIsVoxel `sound good?

Reply:
I would go for a similar approach to what I suggested above. Otherwise I'm in the same spot, not a fan of `Test` but without a better alternative.  So I will leave it up to you! =D